### ISSUE TITLE
Segmentation Fault with myloader command on RHEL7

### ISSUE THREAD

**Issue by maahutch** [maxbube/mydumper#276]
I'm trying to restore a mysql dump using the myloader command. The tables are created but when it tries to load the data it exits with a message 'Segmentation Fault'. I've tried scaling down to just one thread but get the same issue. Other people are using the same server so is this just a resource constraint issue? If I know the size of my data, is there a way to estimate how much memory I need to complete the restore operation? 

Version Info:

mydumper 0.9.5, built against MySQL 5.7.21-21

mysql  Ver 8.0.22 for Linux on x86_64 (MySQL Community Server - GPL)

Operating System: Red Hat Enterprise Linux
CPE OS Name: cpe:/o:redhat:enterprise_linux:7.8:GA:server
Kernel: Linux 3.10.0-1127.19.1.el7.x86_64
Architecture: x86-64

**Comment by davidducos** [comment]
Hi @maahutch, 

Can you send the list of the files that you want to import? I saw in the past some users, that changed the filename and that causes the issue that you have.

**Comment by maahutch** [comment]
This is the command I'm runnning: 
myloader -d '/N/project/mag/uspto_june_2020/PatentsView_20200630/' -B uspto_2020 -v 3 -t 16 -h localhost -u maahutch -a

I've attached a list with the contents of the dump file. 
[dump_file_contents.txt](https://github.com/maxbube/mydumper/files/5495369/dump_file_contents.txt)

Could the issue be that the database I'm restoring to has a different name than the original?

**Comment by fredricj** [comment]
The issue is likely add_index.sql and drop_index.sql. myloader does not gracefully handle sql files with only one '.' in their name, see #210

**Comment by davidducos** [comment]
@maahutch does @fredricj comment helped you?

**Comment by maahutch** [comment]
It turned out the issue was the database name. I didn't realize I had to restore the dump file into a database with the same name as the original. I had created a new database with a new name and was trying to restore to that. Once I renamed my empty database the restore worked without a problem. 

It was the 'Segmentation Fault' error message that made me think something more complex had gone wrong.
